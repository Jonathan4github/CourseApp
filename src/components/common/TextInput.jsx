import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  error: PropTypes.string
};

const defaultProps = {
  value: '',
  error: '',
  placeholder: ''
};

const TextInput = ({
  name, label, onChange, onFocus, placeholder, value, error
}) => {
  let wrapperClass = 'field';
  if (error && error.length > 0) {
    wrapperClass += ' has-error';
  }
  return (
    <div className="form-group">
      <label htmlFor={name}>{label}</label>
      <div className={wrapperClass}>
        <input
          type="text"
          className="form-control"
          name={name}
          id={name}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          onFocus={onFocus}
          autoComplete="off" />
      </div>
      {error && <div className="field-error">{error}</div>}
    </div>
  );
};

TextInput.propTypes = propTypes;
TextInput.defaultProps = defaultProps;

export default TextInput;
