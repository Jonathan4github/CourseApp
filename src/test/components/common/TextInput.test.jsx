/* eslint-disable */
import TextInput from '../../../components/common/TextInput';

const setup = (error) => {
  const props = {
    name: '',
    label: '',
    onChange: jest.fn(),
    onFocus: jest.fn(),
    placeholder: '',
    value: '',
    error
  };

  return mount(<TextInput {...props} />);
};

let mountWrapper;

describe('<TextInput /> component test', () => {
  it('renders component without crashing', () => {
    mountWrapper = setup('');
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('should not render error divs when no error occur', () => {
    mountWrapper = setup('');
    expect(mountWrapper.find('.has-error').length).toEqual(0);
    expect(mountWrapper.find('.field-error').length).toEqual(0);
  });

  it('renders error divs when error occur', () => {
    mountWrapper = setup('field is required');
    expect(mountWrapper.find('.has-error').length).toEqual(1);
    expect(mountWrapper.find('.field-error').length).toEqual(1);
  });
});
