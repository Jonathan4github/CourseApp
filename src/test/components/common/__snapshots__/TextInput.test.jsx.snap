// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> component test renders component without crashing 1`] = `
<TextInput
  error=""
  label=""
  name=""
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  placeholder=""
  value=""
>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <div
      className="field"
    >
      <input
        autoComplete="off"
        className="form-control"
        id=""
        name=""
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        placeholder=""
        type="text"
        value=""
      />
    </div>
  </div>
</TextInput>
`;
